src/Components/Web.Extensions/src/InputFile/RemoteBrowserFileStream.cs

```diff
@@ -0,0 +1,108 @@
+            : base(file)
+        {
+            _jsRuntime = jsRuntime;
+            _inputFileElement = inputFileElement;
+            _maxChunkSize = maxChunkSize;
+            _chunkSequence = new PreFetchingSequence(
+                FetchEncodedChunk,
+                (File.Size + _maxChunkSize - 1) / _maxChunkSize,
+                Math.Max(1, maxBufferSize / _maxChunkSize)); // Degree of parallelism on fetch.
+            _currentChunkDecodingBuffer = new byte[_maxChunkSize];
+        }
+
+        protected override async ValueTask<int> CopyFileDataIntoBuffer(long sourceOffset, Memory<byte> destination, CancellationToken cancellationToken)
+        {
+            var totalBytesCopied = 0;
+
+            while (destination.Length > 0)
+            {
+                // If we don't yet have a chunk, or it's fully consumed, get the next one.
+                if (!_currentChunk.HasValue || _currentChunkDecodingBufferConsumedLength == _currentChunk.Value.LengthBytes)
+                {
+                    // If we've already read some data, and the next chunk is still pending,
+                    // then just return now rather than awaiting.
+                    if (totalBytesCopied > 0 && _chunkSequence.TryPeekNext(out var nextChunk) && !nextChunk.Base64.IsCompleted)
+                    {
+                        break;
+                    }
+
+                    _currentChunk = _chunkSequence.ReadNext(cancellationToken);
+
+                    var currentEncodedChunk = await _currentChunk.Value.Base64;
+
+                    DecodeChunkToBuffer(currentEncodedChunk, _currentChunkDecodingBuffer, 0, _currentChunk.Value.LengthBytes);
+
+                    _currentChunkDecodingBufferConsumedLength = 0;
+                }
+
+                // How much of the current chunk can we fit into the destination?
+                var numUnconsumedBytesInChunk = _currentChunk.Value.LengthBytes - _currentChunkDecodingBufferConsumedLength;
+                var numBytesToTransfer = Math.Min(numUnconsumedBytesInChunk, destination.Length);
+
+                if (numBytesToTransfer == 0)
+                {
+                    break;
+                }
+
+                // Perform the copy.
+                new Memory<byte>(_currentChunkDecodingBuffer, _currentChunkDecodingBufferConsumedLength, numBytesToTransfer).CopyTo(destination);
+                destination = destination.Slice(numBytesToTransfer);
+                _currentChunkDecodingBufferConsumedLength += numBytesToTransfer;
+                totalBytesCopied += numBytesToTransfer;
+            }
+
+            return totalBytesCopied;
+        }
+
+        private EncodedFileChunk FetchEncodedChunk(long index, CancellationToken cancellationToken)
+        {
+            var sourceOffset = index * _maxChunkSize;
+            var chunkLength = (int)Math.Min(_maxChunkSize, File.Size - sourceOffset);
+            var task = _jsRuntime.InvokeAsync<string>(
```

Comment:
Should we special case this in SSB to avoid the string/base64 serialization all-together and allow for binary transfers directly on the BlazorHub? (This is more of an open question). It is definitely more work, and I'm ok if we consider it out of scope, but at the same time this will be a 25% bandwidth improvement over the current b64 serialization format.

Reply:
This was something @SteveSandersonMS also brought up, but I believe we decided it was out of scope for this PR. Perhaps we could make a new issue for this.